lib/webdriveragent.js

```diff
@@ -80,12 +110,15 @@ class WebDriverAgent {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
+      this.setupNoSessionProxy();
       this.setupProxy(sessionId);
       return this.webDriverAgentUrl;
     }
 
     log.info('Launching WebDriverAgent on the device');
 
+    this.setupNoSessionProxy();
```

Comment:
unfortunately it is not possible to combine proxies init into single method, since noSessionProxy is used for getting driver status to detect whether WDA is running after xcodebuild has been executed. And the _jwproxy_ can only be initialised after _sessionId_ is known (which is known only after session init)